### PR TITLE
8308092: Replace NULL with nullptr in gc/x

### PR DESCRIPTION
**Pull request by stefank**
Replace NULL with nullptr in gc/x. We've already done this work for Generational ZGC, but left it for the Singlegen ZGC code.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308092](https://bugs.openjdk.org/browse/JDK-8308092): Replace NULL with nullptr in gc/x

### Reviewers
 * [Erik Österlund](https://openjdk.org/census#eosterlund) (@fisk - **Reviewer**)
 * [Axel Boldt-Christmas](https://openjdk.org/census#aboldtch) (@xmas92 - Committer)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13984/head:pull/13984` \
`$ git checkout pull/13984`

Update a local copy of the PR: \
`$ git checkout pull/13984` \
`$ git pull https://git.openjdk.org/jdk.git pull/13984/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13984`

View PR using the GUI difftool: \
`$ git pr show -t 13984`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13984.diff">https://git.openjdk.org/jdk/pull/13984.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13984#issuecomment-1547703490)

### CODE REVIEW

**Review by fisk — APPROVED** [review]
Looks good.

**Review by xmas92 — APPROVED** [review]
lgtm.

File: src/hotspot/share/gc/x/xBarrier.inline.hpp [hunk 1]
```diff
@@ -226,7 +226,7 @@ inline bool XBarrier::during_relocate() {
 // Load barrier
 //
 inline oop XBarrier::load_barrier_on_oop(oop o) {
-  return load_barrier_on_oop_field_preloaded((oop*)NULL, o);
+  return load_barrier_on_oop_field_preloaded((oop*)nullptr, o);
```
Comment (dholmes-ora) [inline src/hotspot/share/gc/x/xBarrier.inline.hpp:229]: Casts should not be needed on `nullptr`.

**Review by dholmes-ora — APPROVED** [review]
Looks good. There are 5 casts that should hopefully no longer be needed.

Thanks.

**Comment by jdksjolen** [comment]
Thank you for this!

File: src/hotspot/share/gc/x/xBarrier.inline.hpp (hunk 1 above)
Comment (stefank) [reply src/hotspot/share/gc/x/xBarrier.inline.hpp:229]: They are needed to disambiguate the two overloaded functions that take `volatile oop*` and  `volatile narrowOop*` respectively.

**Comment by stefank** [comment]
Thanks for reviewing!

**Pull request closed without merging**